indexer/lib/advertisement-walker.js

```diff
@@ -169,7 +169,10 @@ export async function processNextAdvertisement ({
       state.status = `Walking the advertisements from ${state.head}, next step: ${state.tail}`
     }
 
-    const indexEntry = entry.pieceCid ? entry : undefined
+    if (entriesFetchError) {
+      state.entriesNotRetrievable = (state.entriesNotRetrievable ?? 0) + 1
+    }
+    const indexEntry = entry.pieceCid && entry.payloadCid ? entry : undefined
```

Comment:
Do you mean

```suggestion
    const indexEntry = (entry.pieceCid && entry.payloadCid) ? entry : undefined
```

or

```suggestion
    const indexEntry = entry.pieceCid && (entry.payloadCid ? entry : undefined)
```

Could you please make this explicit?